crates/revm/src/inspector/eip3155.rs

```diff
@@ -22,6 +22,8 @@ pub struct TracerEip3155 {
     refunded: i64,
     mem_size: usize,
     skip: bool,
+    include_memory: bool,
+    memory: Option<Vec<u8>>,
```

Comment:
Onee more thing, it would be good to have `String` here so we don't copy memory multiple times.

Now we are doing clone in `step` and clone into String in `step_end`

Reply:
I've changed it to String now, however my change did not improve the performance. I guess the `self.memory.take()` method also copies it internally. Without the `.take()` I've got the error "cannot move out of `self.memory` which is behind a mutable reference" and didn't manage to fix it in a better way.

Reply:
`take()` does not copy full string, it is the right approach. Copying memory on every step is a heavy operation, not a lot we could do there. There are few optimisation that we could do but they require introspection of opcodes that change memory, definitely out of scope for this PR.